File: 백준/레벨Silver III/3273. 두 수의 합/두 수의 합.py
n = int(input().rstrip())
arr = list(map(int, input().split()))
x = int(input().rstrip())

arr.sort()

cnt, i, j = 0, 0, n-1
while i < j:
    curr = arr[i] + arr[j]
    if curr < x:
        i += 1
    elif curr > x:
        j -= 1
    else:
        cnt += 1
        i += 1
        j -= 1

print(cnt)

File: 백준/레벨Silver III/3273. 두 수의 합/두 수의 합.ts
import fs = require('fs');

const lines = fs.readFileSync(0, 'utf-8').trim().split('\n');
const n = parseInt(lines[0]);
const arr = lines[1].split(' ').map(Number);
const x = parseInt(lines[2]);

arr.sort((a: number, b: number) => a - b);

let cnt = 0,
  i = 0,
  j = n - 1;
while (i < j) {
  const curr = arr[i] + arr[j];
  if (curr < x) i++;
  else if (curr > x) j--;
  else {
    cnt++;
    i++;
    j--;
  }
}

console.log(cnt);